File: src/Wolf.java
public class Wolf extends Canine {

    Wolf(){
        super(true, "Omnivore", "Suburbs", true, "MoonMoon", 4, false);
    }

    /**
     *  Constructor with only Animal parameters
     * @param awake:boolean - Is the animal awake?
     * @param diet:String - What does the animal eat?
     * @param habitat:String - Where does the animal live?
     * @param hungry:boolean - Is the animal hungry?
     * @param name:String - The name of the animal
     * @param numLegs:int - How many legs does the animal have?
     */

    Wolf(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs){
        super(awake, diet, habitat, hungry, name, numLegs);
    }

    /**
     *  Constructor with Animal and Canine parameters
     * @param awake:boolean - Is the animal awake?
     * @param diet:String - What does the animal eat?
     * @param habitat:String - Where does the animal live?
     * @param hungry:boolean - Is the animal hungry?
     * @param name:String - The name of the animal
     * @param numLegs:int - How many legs does the animal have?
     *
     * @param domesticated:boolean - Is this animal domesticated
     */
    Wolf(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs, boolean domesticated){
        super(awake, diet, habitat, hungry, name, numLegs, domesticated);
    }

    /**
     * Override roam for Wolves
     */
    @Override
    public void roam() {
        double randomNumber = Math.random();
        if(randomNumber < 0.7) {
            System.out.println(name + " the Wolf: hides in his cave.");
        } else {
            System.out.println(name + " the Wolf: looks for some meat to eat.");
        }

    }

    /**
     * Override make noise for Wolves
     */
    @Override
    public void makeNoise() {
        double rand = Math.random();
        if (rand < 0.2) {
            System.out.println(name + " the Wolf: howls");
        } else {
            System.out.println(name + " the Wolf: aggressively growls.");
        }
    }


}

File: Project2/Wolf.py
import random
from Canine import Canine

class Wolf(Canine):

  '''
    Generic initializer for Wolf
    @param name:String, default:'Wale'
    @param awake:Boolean, default:False
    @param hungry:Boolean, deault:True
    @param num_legs:Int, default:4
  '''
  def __init__(self, name='Wale', awake=False, hungry=True, num_legs=4):
    super(Wolf, self).__init__(name, awake, hungry, num_legs)

  '''
    Wolf has a 50% chance to either howl or agressively growl
  '''
  def make_noise(self):
    animal_type = type(self).__name__
    random_number = random.random()
    if random_number < 0.5:
      print('{} the {} howled.'.format(self.name, animal_type))
    else:
      print('{} the {} aggressively growled.'.format(self.name, animal_type))
      

